src/prefect/client/orchestration.py

```diff
@@ -2594,22 +2594,36 @@ async def send_worker_heartbeat(
         work_pool_name: str,
         worker_name: str,
         heartbeat_interval_seconds: Optional[float] = None,
-    ):
+        get_worker_id: bool = False,
+    ) -> Optional[UUID]:
         """
         Sends a worker heartbeat for a given work pool.
 
         Args:
             work_pool_name: The name of the work pool to heartbeat against.
             worker_name: The name of the worker sending the heartbeat.
+            return_id: Whether to return the worker ID.
         """
-        await self._client.post(
+
+        if get_worker_id:
+            return_dict = {"return_id": get_worker_id}
+        else:
+            return_dict = {}
+
+        resp = await self._client.post(
             f"/work_pools/{work_pool_name}/workers/heartbeat",
             json={
                 "name": worker_name,
                 "heartbeat_interval_seconds": heartbeat_interval_seconds,
-            },
+            }
+            | return_dict,
         )
 
+        if get_worker_id and resp.status_code == 200:
+            return UUID(resp.text)
+        else:
+            return None
```

Comment:
I think we might want to include a check for whether the client is connected to Cloud in this method. If someone is using this method directly with an self-hosted server, it'd cause problems.

Reply:
By default `/heartbeat` should only send 204s not 200s but I agree it's probably better to be safe then sorry!